tests/e2e/e2e_test.go

```diff
@@ -155,15 +140,17 @@ func testAttachWriteReadDetach(volumeID, volName, nodeID string, readOnly bool)
 		VolumeCapability:  stdVolCap[0],
 	})
 	Expect(err).To(BeNil(), "NodePublishVolume failed with error")
-	//err = testutils.ForceChmod(instance, filepath.Join("/tmp/", volName), "777")
-	//Expect(err).To(BeNil(), "Chmod failed with error")
-	testFileContents := "test"
+
 	if !readOnly {
 		// Write a file
+		testFileContents := []byte("sample content")
 		testFile := filepath.Join(publishDir, "testfile")
-		//err = testutils.WriteFile(instance, testFile, testFileContents)
-		err := ioutil.WriteFile(testFile, []byte(testFileContents), 0644)
+		err := ioutil.WriteFile(testFile, testFileContents, 0644)
 		Expect(err).To(BeNil(), "Failed to write file")
+		// Read the file and check if content is correct
+		data, err := ioutil.ReadFile(testFile)
+		Expect(err).To(BeNil(), "Failed to read file")
+		Expect(data).To(Equal(testFileContents), "File content is incorrect")
 	}
 
 	// Unmount Disk
```

Comment:
Should move the umount to be right after mount disk succeeds. So that any failures in side file read/write won't cause file is not umounted.

Reply:
Good catch, thanks.